midterm-2023-11-06/code/5163-mihaela-milosevska/script.ts

```diff
@@ -0,0 +1,82 @@
+interface album{
+    name: string;
+    year; number;
+}
+
+interface band {
+    id: number;
+    name: string;
+    formed: number;
+    location: string;
+    genre: string;
+    members: string;
+    albums: number;
+}
+type BandSorter = (first: Band, second: Band) => number;
+
+async function getBands(): Promise<Band[]> {
+    const url = "https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json";
+    const response = await axios.get(url);
+    return response.data.metalBands;
+}
+
+function formatBand(band: Band): Record<string, unknown> {

```

Comment:
The signature of the function is actually the wrong way round - it returns a `band` object, it accepts the data object from the API. The mapping is implemented ok, so +15